sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -74,7 +74,8 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
       case (TIMESTAMP_LTZ, Nil) => TimestampType
       case (STRING, Nil) =>
         typeCtx.children.asScala.toSeq match {
-          case Seq(_) => StringType
+          case Seq(_) =>
+            StringType(CollationFactory.collationNameToId(SqlApiConf.get.defaultCollation))
```

Comment:
a safer approach is to still return the object `StringType` if the default collation is `UTF8_BINARY_COLLATION_ID`

Reply:
Yeah, I was concerned about this too, as some tests did fail because of this change.